wicket-core/src/main/java/org/apache/wicket/markup/html/panel/PanelMarkupSourcingStrategy.java

```diff
@@ -33,9 +33,28 @@
  */
 public class PanelMarkupSourcingStrategy extends AssociatedMarkupSourcingStrategy
 {
+	private static final PanelMarkupSourcingStrategy PANEL_INSTANCE = new PanelMarkupSourcingStrategy(
+		false);
+	private static final PanelMarkupSourcingStrategy BORDER_INSTANCE = new PanelMarkupSourcingStrategy(
+		true);
+
 	// False for Panel and true for Border components.
 	private final boolean allowWicketComponentsInBodyMarkup;
```

Comment:
Is it a good moment to re-work this boolean to an enum ?
With an enum it would be possible to have a third/forth/... state if needed. And it is more clear than `true`/`false`

Reply:
@martin-g: I experimentally refactored this in https://github.com/theigl/wicket/pull/1. Could you take a look? I couldn't think of a good name for the new enum. I'm not sure this refactoring is really worth it.

Reply:
> I couldn't think of a good name for the new enum.

`Type` ?

Reply:
I'd drop the usage of the `boolean` in the enum. Just compare the variants.

> I'm not sure this refactoring is really worth it.

I think it is a good idea, but I am not keen on it.
Thanks for trying it anyway!